tests/e2e/magic_functions/functions/c_read.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "mg_procedure.h"
+
+static mgp_func_result *return_function_argument(struct mgp_list *args, mgp_func_context *ctx,
+                                                 struct mgp_memory *memory) {
+  mgp_func_result *result{};
+
+  mgp_value *value{};
+  auto err_code = mgp_list_at(args, 0, &value);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_error("Failed to fetch list!", memory, &result);
+    return result;
+  }
+
+  err_code = mgp_func_result_value(value, memory, &result);
+  if (err_code != MGP_ERROR_NO_ERROR) {
+    mgp_func_result_error("Failed to construct return value!", memory, &result);
+    return result;
+  }
+  return result;
+}
+
+double get_element_from_arg(struct mgp_list *args, int index) {
+  mgp_value *value{};
+  mgp_list_at(args, index, &value);
```

Comment:
I think it would be nice to check the return type here as things can go really wrong if something doesn't match the expected structure/type.